### ISSUE TITLE
[octomap] deprecated declarations as error

### ISSUE THREAD

**Issue by acxz** [acxz/pkgbuilds#190]
```
In file included from /home/acxz/vcs/git/github/acxz/pkgbuilds/octomap/src/octomap-1.9.8/octomap/include/octomap/OcTreeBaseImpl.h:83,
                 from /home/acxz/vcs/git/github/acxz/pkgbuilds/octomap/src/octomap-1.9.8/octomap/include/octomap/OccupancyOcTreeBase.h:44,
                 from /home/acxz/vcs/git/github/acxz/pkgbuilds/octomap/src/octomap-1.9.8/octomap/include/octomap/OcTree.h:38,
                 from /home/acxz/vcs/git/github/acxz/pkgbuilds/octomap/src/octomap-1.9.8/octomap/src/OcTree.cpp:34:
/home/acxz/vcs/git/github/acxz/pkgbuilds/octomap/src/octomap-1.9.8/octomap/include/octomap/OcTreeIterator.hxx:42:39: error: ‘template<class _Category, class _Tp, class _Distance, class _Pointer, class _Reference> struct std::iterator’ is deprecated [-Werror=deprecated-declarations]
   42 |     class iterator_base : public std::iterator<std::forward_iterator_tag, NodeType>{
      |                                       ^~~~~~~~
In file included from /usr/include/c++/12.1.0/bits/stl_algobase.h:65,
                 from /usr/include/c++/12.1.0/list:60,
                 from /home/acxz/vcs/git/github/acxz/pkgbuilds/octomap/src/octomap-1.9.8/octomap/include/octomap/OccupancyOcTreeBase.h:38:
/usr/include/c++/12.1.0/bits/stl_iterator_base_types.h:127:34: note: declared here
  127 |     struct _GLIBCXX17_DEPRECATED iterator
      |                                  ^~~~~~~~
cc1plus: all warnings being treated as errors
```

**Comment by acxz** [comment]
upstream issue: https://github.com/OctoMap/octomap/issues/372

**Comment by acxz** [comment]
patched with [5f59f42](https://github.com/acxz/pkgbuilds/commit/5f59f424d94e1bbfa9ed5b515168315ac06cd58e)

**Comment by acxz** [comment]
Resolved with [fadda6f](https://github.com/acxz/pkgbuilds/commit/fadda6f9b5a0d4e02679ba372280a8745ccdc257)
